spec/requests/answer_controller_spec.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe DiscourseSolved::AnswerController do
+  fab!(:user)
+  fab!(:staff_user) { Fabricate(:admin) }
+  fab!(:category)
+  fab!(:topic) { Fabricate(:topic, category: category) }
+  fab!(:p) { Fabricate(:post, topic: topic) }
+  fab!(:solution_post) { Fabricate(:post, topic: topic) }
+
+  before do
+    SiteSetting.solved_enabled = true
+    SiteSetting.allow_solved_on_all_topics = true
+    category.custom_fields[DiscourseSolved::ENABLE_ACCEPTED_ANSWERS_CUSTOM_FIELD] = "true"
+    category.save_custom_fields
+
+    # Give permission to accept solutions
+    user.update!(trust_level: 1)
+
+    # Make user the topic creator so they can accept answers
+    topic.update!(user_id: user.id)
+  end
+
+  describe "#accept" do
+    context "with default rate limiting" do
+      it "applies rate limits to regular users" do
+        sign_in(user)
+
+        # Should be rate limited
+        RateLimiter.any_instance.expects(:performed!).raises(RateLimiter::LimitExceeded.new(60))
+        post "/solution/accept.json", params: { id: solution_post.id }
+        expect(response.status).to eq(429)
+      end
+
+      it "does not apply rate limits to staff" do
+        sign_in(staff_user)
+
+        post "/solution/accept.json", params: { id: solution_post.id }
+        expect(response.status).to eq(200)
+      end
+    end
+
+    context "with plugin modifier" do
+      it "allows plugins to bypass rate limiting" do
+        sign_in(user)
+        # Create a mock plugin instance with enabled? method
+        plugin_instance = Object.new
```

Comment:
Instead of doing a fake plugin_instance can you follow the pattern in this spec? 
https://github.com/discourse/discourse/blob/ae4ee7754f0d47edd5e21fefc220462f6e5f1c22/spec/models/site_spec.rb#L254-L269